src/main/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSink.java

```diff
@@ -33,6 +34,12 @@
      */
     @Override
     public void recordAggregateData(final PeriodicData data) {
+        if (_whitelistDimensions.isPresent()) {
+            if (!_whitelistDimensions.get().containsAll(data.getDimensions().keySet())) {
```

Comment:
Definitely the latter - intending to use this on "legacy" sinks like telemetry and monitord. It is this bit that I don't yet see as as need: "where we want to only allow aggregations with some defined set, and all of that set" - I don't see the need to have this filter _require_ that the _entire_ set is present. I think I need some clarification here...

I have no idea what this means, I'm sorry! "the image implementation in MAD has already been unified and the implementation here will be too." Perhaps a voice chat or in-person would be useful.